corehq/form_processor/management/commands/archive_all_forms_for_user_in_domain.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+
+from io import open
+
+from django.core.management.base import BaseCommand
+
+from casexml.apps.case.cleanup import rebuild_case_from_forms
+from casexml.apps.case.xform import get_case_ids_from_form
+from corehq.apps.users.models import CouchUser
+from corehq.form_processor.backends.sql.dbaccessors import LedgerAccessorSQL
+from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+from corehq.form_processor.models import RebuildWithReason
+from corehq.util.log import with_progress_bar
+from corehq.form_processor.interfaces.processor import FormProcessorInterface
+from corehq.form_processor.parsers.ledgers.form import get_ledger_references_from_stock_transactions
+
+
+class Command(BaseCommand):
+    help = """
+        Bulk archive forms for user on domain.
+        First archive all forms and then rebuild corresponding cases
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('user_id')
+        parser.add_argument('domain')
+
+    def handle(self, user_id, domain, **options):
+        user = CouchUser.get_by_user_id(user_id)
+        form_accessor = FormAccessors(domain)
+
+        # ordered with latest form's id on top
+        form_ids = form_accessor.get_form_ids_for_user(user_id)
+        forms = [f for f in form_accessor.get_forms(form_ids) if f.is_normal]
+        print("Found %s normal forms for user" % len(form_ids))
+
+        case_ids_to_rebuild = set()
+        for form in forms:
+            case_ids_to_rebuild.update(get_case_ids_from_form(form))
+        print("Found %s cases that would need to be rebuilt" % len(case_ids_to_rebuild))
+
+        # archive forms
+        print("Starting with form archival")
+        with open("forms_archived.txt", "w+b") as forms_log:
+            for form in with_progress_bar(forms):
+                forms_log.write("%s\n" % form.form_id)
+                form.archive(rebuild_cases=False)
+
+        # removing data
+        for xform in with_progress_bar(forms):
+            refs_to_rebuild = get_ledger_references_from_stock_transactions(xform)
```

Comment:
This is repeating the work done in `get_case_ids_from_form`. If it were a cheap operation it would be fine but it's not super cheap so doing it over seems inefficient. You could just use the same set of case_ids returned by `get_case_ids_from_form` or you could split it out:
```
case_ids = set(cu.id for cu in get_case_updates(xform))
ledger_case_ids = get_case_ids_from_stock_transactions(xform)
```

Reply:
thanks, updated.